clients/drcachesim/tracer/raw2trace.cpp

```diff
@@ -1832,12 +1845,11 @@ raw2trace_t::adjust_and_emit_rseq_buffer(raw2trace_thread_data_t *tdata, addr_t
         // up on distinguishing a side exit whose target is the end PC fall-through from
         // a completion.
         log(4, "Rseq completed normally\n");
-    }
-    // We actually won't see the handler PC as the rseq abort event has the
-    // commit PC and we have it call here and don't wait for a subsequent
-    // instr (or signal interruption) to find the handler PC; so here it's
-    // not really the "next_pc" but the "exit_pc".
-    else if (next_pc == tdata->rseq_commit_pc_) {
+    } else if ((at_abort_marker && next_pc == tdata->rseq_handler_pc_) ||
```

Comment:
Is this why we want to have TRACE_MARKER_TYPE_RSEQ_HANDLER? Instead of having just `at_abort_marker` here?

Reply:
Right but if we're at the abort marker we have the handler value as you pointed out so trying the earlier suggestion.